### DISCUSSION TITLE
Cannot run on MAC 10.12

### DISCUSSION THREAD

**Discussion by dongjian** [fltk-rs/fltk-rs#960] [Q&A]
## Describe the bug
I compiled on Mac 11.6 and it worked fine.

Cargo.toml:
[dependencies]
FLTK = "1.2.3"

build.rs:
println! (" cargo: rustc - env = MACOSX_DEPLOYMENT_TARGET = 10.8 ");

But I copied the running program to Mac 10.12 and got an error
dyld:lazy symbol binding failed: symbol not found: _objc_alloc_init

## Screenshots

![image](https://user-images.githubusercontent.com/2841022/139410810-1a49e5b6-a3a8-470c-9a84-980458d96bfd.png)

**Comment by MoAlyousef** [comment]
You can try setting the MACOSX_DEPLOYMENT_TARGET env variable before even calling cargo build since CMake also reads it, and setting it in your build.rs works only when your crate is built, not with the other crates. 

Generally though, the deployment target has limitations, especially with end of life macos versions. MacOS older versions aren’t assured to run binaries built with newer versions, they don’t have forward-compatibility, especially when it comes to libc and libobjc. Usually you would build on the older version and would be able to run on the newer version, which have backward-compatibility. 

https://stackoverflow.com/questions/65060022/macos-app-bundle-doesnt-work-on-lower-macos-versions-than-what-its-built-on

**Reply by dongjian** [reply]
Thanks for your reply. I recompiled the project on Mac 10.12. But something went wrong.
![image](https://user-images.githubusercontent.com/2841022/139588340-8a0796f1-a37f-4af4-abf6-15921d1e0b72.png)



After I comment out the code associated with fltk_webView, the project works fine.
By the way FileDialog is not clickable on the left on this system version.
Do you need me to mention a related issue in FLtk_webview?
Thank you for your help

**Reply by MoAlyousef** [reply]
The current version of [webview](https://github.com/webview/webview), doesn't seem to support older MacOS versions. Looking at the Tauri, tracker, the same issue appears here:
https://github.com/tauri-apps/tauri/issues/2732
Which mentions also another related issue.
The FileDialog only spawns a MacOS native file dialog, it might be an MacOS SDK issue.

**Reply by dongjian** [reply]
Thank you for your quick and patient reply. I will tell the client that we support the minimum version 10.15 for Mac.
